TradeItIosTicketSDK2/TradeItAlertManager.swift

```diff
@@ -4,67 +4,65 @@ import UIKit
     var linkedBrokerManager = TradeItLauncher.linkedBrokerManager
     var linkBrokerUIFlow = TradeItLinkBrokerUIFlow(linkedBrokerManager: TradeItLauncher.linkedBrokerManager)
 
-    func showGenericError(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
-                            onViewController viewController: UIViewController,
-                                             onFinished: () -> Void = {}) {
-        let alertTitle = tradeItErrorResult.shortMessage ?? ""
-        let messages = (tradeItErrorResult.longMessages as? [String]) ?? []
-        let alertMessage = messages.joinWithSeparator(" ")
-        let alertActionTitle = "OK"
+    public func showError(error: TradeItErrorResult,
+                          onViewController viewController: UIViewController,
+                          onFinished: () -> Void = {}) {
+        let title = error.shortMessage ?? ""
+        let messages = (error.longMessages as? [String]) ?? []
+        let message = messages.joinWithSeparator(". ")
+        let actionTitle = "OK"
 
-        self.showOn(viewController: viewController,
-                    withAlertTitle: alertTitle,
-                    withAlertMessage: alertMessage,
-                    withAlertActionTitle: alertActionTitle,
+        self.showAlert(onViewController: viewController,
+                              withTitle: title,
+                            withMessage: message,
+                        withActionTitle: actionTitle,
                     onAlertActionTapped: onFinished)
     }
 
-    public func show(tradeItErrorResult tradeItErrorResult: TradeItErrorResult,
-                onViewController viewController: UIViewController,
-                withLinkedBroker linkedBroker: TradeItLinkedBroker,
-                                 onFinished : () -> Void) {
+    public func showRelinkError(error: TradeItErrorResult,
```

Comment:
This will be a larger refactor. I've put a story in it. We are using it in lots of places which may get other errors like connection failures as well. Now all of those places will need to determine the error type and call `promptUserToRelink`.